formulas/functions/reference.js

```diff
@@ -343,6 +343,41 @@ const ReferenceFunctions = {
     return result;
   },
 
+  /**
+       * @param {*} range - The data to filter by unique entries.
+       * @returns unique rows in the provided source range, discarding duplicates.
+       *          Rows are returned in the order in which they first appear in the source range.
+       **/
+   UNIQUE: (range) => {
+    try {
+      range = H.accept(range, Types.ARRAY, null, false);
+    } catch (e) {
+      range = H.accept(range);
+      return range;
+    }
+    //Checks to see if range is an array of arrays. If it is not, 
+    //then the first row is Unique to its self and thus can be returned
+    if (typeof range[0] !== "object") {
+      return range;
+    }
+    let seen = new Set();
+    let rv = [];
+    for (let index = 0; index < range.length; index++) {
+      let currValue = range[index];
+      let returnValue = currValue;
```

Comment:
If these vars are the same, why do we have both?

Reply:
B/C I modify currValue to run element wise comparisons. So if range[index] is an array, I convert it to a string via .join(). currValue is my comparison var, while returnValue stored the return value

Reply:
Ok